install/operator/pkg/syndesis/configuration/configuration.go

```diff
@@ -329,47 +329,72 @@ func GetProperties(file string, ctx context.Context, client client.Client, synde
 		return nil, err
 	}
 
-	if client == nil || len(syndesis.Spec.Components.Database.ExternalDbURL) > 0 {
-		return configuration, nil
+	//
+	// If an external database has been defined then reset properties appropriately
+	//
+	if client != nil && len(syndesis.Spec.Components.Database.ExternalDbURL) > 0 {
+		if err := configuration.externalDatabase(ctx, client, syndesis); err != nil {
+			return nil, err
+		}
 	}
 
-	databaseDeployment := &appsv1.DeploymentConfig{}
-	if err := client.Get(ctx, types.NamespacedName{Namespace: syndesis.Namespace, Name: "syndesis-db"}, databaseDeployment); err == nil {
+	//
+	// Only required if not using an external database and need to check
+	// on whether the deployment config is already is upgrade mode
+	//
+	if client != nil && len(syndesis.Spec.Components.Database.ExternalDbURL) <= 0 {
+		databaseDeployment := &appsv1.DeploymentConfig{}
+		if err := client.Get(ctx, types.NamespacedName{Namespace: syndesis.Namespace, Name: "syndesis-db"}, databaseDeployment); err != nil {
```

Comment:
This seemed an innocent change but in fact uncovers a problem with the runtime client. It doesn't support DeploymentConfig. Looking at [L337](https://github.com/syndesisio/syndesis/pull/8314/files#diff-8f63b1158cd0c8188207de68b0013e69L337) the condition was `if err == null` so if the `client.Get(...)` had no error then do stuff. The change turns it around so that if there is an error then flag it. Therefore, the following error was being swallowed and ignored:
```
error: no kind is registered for the type v1.DeploymentConfig in scheme "k8s.io/client-go/kubernetes/scheme/register.go:69"
```
At some point, this started happening because this code did its job of maintaining an upgrade. So at some point it stopped working??

@lgarciaaco